vault/provider.go

```diff
@@ -869,15 +869,14 @@ func ReadContextWrapper(f schema.ReadContextFunc) schema.ReadContextFunc {
 // wrapped schema.CreateContextFunc.
 func MinVersionCheckWrapper(f schema.CreateContextFunc, minVersion string) schema.CreateContextFunc {
 	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-		client, e := provider.GetClient(d, meta)
+		mv, e := version.NewVersion(minVersion)
 		if e != nil {
 			return diag.FromErr(e)
 		}
 
-		featureEnabled, currentVersion, err := semver.GreaterThanOrEqual(ctx, client, minVersion)
-		if err != nil {
-			return diag.FromErr(err)
-		}
+		featureEnabled := provider.IsAPISupported(meta, mv)
```

Comment:
nit: this is more at the api level, rather than the feature level, perhaps rename to to something like `apiSupported` or something like that.